pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -117,175 +121,151 @@ func labelMachineSetNodes(client runtimeclient.Client, ms *mapiv1beta1.MachineSe
 		// we need to test against deployments instead so we skip this test.
-		targetMachineSet := machineSetList.Items[0]
 		if ownerReferences := targetMachineSet.GetOwnerReferences(); len(ownerReferences) > 0 {
 			// glog.Infof("MachineSet %s is owned by a machineDeployment. Please run tests against machineDeployment instead", targetMachineSet.Name)
 			g.Skip(fmt.Sprintf("MachineSet %s is owned by a machineDeployment. Please run tests against machineDeployment instead", targetMachineSet.Name))
 		}
 
-		glog.Infof("Create ClusterAutoscaler and MachineAutoscaler objects. Targeting machineSet %s", targetMachineSet.Name)
-		initialNumberOfReplicas := targetMachineSet.Spec.Replicas
-		tenSecondString := "10s"
-		clusterAutoscaler := caov1alpha1.ClusterAutoscaler{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "default",
-				Namespace: e2e.TestContext.MachineApiNamespace,
-			},
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "ClusterAutoscaler",
-				APIVersion: "autoscaling.openshift.io/v1alpha1",
-			},
-			Spec: caov1alpha1.ClusterAutoscalerSpec{
-				ScaleDown: &caov1alpha1.ScaleDownConfig{
-					Enabled:           true,
-					DelayAfterAdd:     &tenSecondString,
-					DelayAfterDelete:  &tenSecondString,
-					DelayAfterFailure: &tenSecondString,
-					UnneededTime:      &tenSecondString,
-				},
-			},
-		}
-		machineAutoscaler := caov1alpha1.MachineAutoscaler{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: fmt.Sprintf("autoscale-%s", targetMachineSet.Name),
-				Namespace:    e2e.TestContext.MachineApiNamespace,
-			},
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "MachineAutoscaler",
-				APIVersion: "autoscaling.openshift.io/v1alpha1",
-			},
-			Spec: caov1alpha1.MachineAutoscalerSpec{
-				MaxReplicas: *initialNumberOfReplicas + 1,
-				MinReplicas: 1,
-				ScaleTargetRef: caov1alpha1.CrossVersionObjectReference{
-					Name:       targetMachineSet.Name,
-					Kind:       "MachineSet",
-					APIVersion: "machine.openshift.io/v1beta1",
-				},
-			},
-		}
-		err = wait.PollImmediate(1*time.Second, e2e.WaitMedium, func() (bool, error) {
-			if err := client.Create(context.TODO(), &clusterAutoscaler); err != nil {
-				if !strings.Contains(err.Error(), "already exists") {
-					glog.Errorf("error querying api for clusterAutoscaler object: %v, retrying...", err)
-					return false, err
-				}
-			}
-			if err := client.Create(context.TODO(), &machineAutoscaler); err != nil {
-				if !strings.Contains(err.Error(), "already exists") {
-					glog.Errorf("error querying api for machineAutoscaler object: %v, retrying...", err)
-					return false, err
-				}
-			}
-			return true, nil
-		})
+		g.By("Create ClusterAutoscaler object")
+		clusterAutoscaler := clusterAutoscalerResource()
+		err = client.Create(context.TODO(), clusterAutoscaler)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		workLoad := newWorkLoad()
+		initialNumberOfReplicas := *targetMachineSet.Spec.Replicas
```

Comment:
can we use `pointer.Int32PtrDerefOr`

Reply:
Wondering if that would help. I would rather panic in case `targetMachineSet.Spec.Replicas == nil` rather than set some default value and then debug unexpected behaviour caused by it for 1h.

Reply:
addressed